lib/adapters/class-searchpress.php

```diff
@@ -232,6 +232,17 @@ public function filter__wp_rest_search_handlers( $search_handlers ) {
 		return $search_handlers;
 	}
 
+	/**
+	 * Query Elasticsearch directly.
+	 *
+	 * @param array $es_args       Elasticsearch query arguments.
+	 *
+	 * @return array Elasticsearch response.
+	 */
+	public function search( array $es_args ): array {
+		return SP_API()->search( wp_json_encode( $es_args ), [ 'output' => ARRAY_A ] );
```

Comment:
Even though SP can return an object or an array, I decided to return arrays here to standardize the return type of the method across adapters.